app/controllers/users_controller.rb

```diff
@@ -16,14 +19,38 @@ def new
   def create
     @user = User.new(user_params)
     if @user.save
-      log_in @user
-      flash[:success] = "Welcome to the Sample App!"
-      redirect_to user_url(@user)
+      @user.send_activation_email
+      flash[:info] = "Please check your email to activate your account."
+      redirect_to root_url
     else
       render 'new'
     end
   end
 
+  def edit 
+    @user = User.find(params[:id])
+  end 
+
+  def update 
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else 
+      render 'edit'
+    end 
+  end 
+
+  def destroy 
+    User.find(params[:id]).destroy 
+    flash[:success] = "User deleted"
+    redirect_to users_url 
+  end 
```

Comment:
Yes, so far I've only limited destroying to the admin (who has access to the destroy links in the view.) So, I'll go ahead and include that restriction in the controller.